internal/pkg/api/server/templates/dependencies/dependencies.go

```diff
@@ -96,6 +99,38 @@ func (v *container) RepositoryManager() (*repository.Manager, error) {
 	return v.repositoryManager, nil
 }
 
+func (v *container) TemplateRepository(definition model.TemplateRepository, _ model.TemplateRef) (*repository.Repository, error) {
+	// Get manager
+	manager, err := v.RepositoryManager()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get git repository
+	gitRepository, err := manager.Repository(definition)
+	if err != nil {
+		return nil, err
+	}
+
+	// Acquire read lock and release it after request,
+	// so pull cannot occur in the middle of the request.
+	gitRepository.RLock()
+	go func() {
+		<-v.ctx.Done()
+		gitRepository.RUnlock()
+	}()
```

Comment:
V predchadzajucom PR som pridaval metodu `gitRepository.CallWithFs`, kedze je tam potrebny vyriesit zamykanie.
To sa ukazalo ako nepouzitelne, ked FS potrebujem vratit ako sucast `repository`, vid nizsie `return`.

Preto som pridal public metody `RLock`, a `RUnlock`.
Po dokonceni requestu za zavola `RUnlock` a teda `pull` nemoze nastat nikdy v polovcke requestu.

Pull vola `Lock` metodu, a teda ked pride poziadavka na `pull` (z ticker), tak dobehnu requesty, ktore su uz spustene, ... a ostatne budu cakat na tomto mieste kym pull dobehne.